src/main/kotlin/com/exactpro/th2/read/file/common/AbstractFileReader.kt

```diff
@@ -749,7 +725,7 @@ abstract class AbstractFileReader<T : AutoCloseable>(
             || (if (configuration.disableFileMovementTracking) fileHolder.path != path else !fileTracker.isSameFiles(fileHolder.path, path))
     }
 
-    protected class FileHolder<T : AutoCloseable>(
+    public class FileHolder<T : AutoCloseable>(
```

Comment:
TransportDefaultFileReader.Builder and PrototDefaultFileReader.Builder should be public, because of this DefaultFileReader.Builder also should be public (because superclass of 'public' class can't be 'protected' in Kotlin), because of this FileHolder also should be public as since it's exposed by protected member of nested Builder class and can be accessed from outside of AbstractFileReader class or it's subclasses.

this code just can't be compiled if FileHolder is not 'public'.

Reply:
Can you try to make DefaultFileReader.Buildet sealed class? I think this should solve the compilation problem

Reply:
I've tried this. It produces the same compile time errors.